hicommonwealth/commonwealth
# Fix members page infinite loop

## Link to Issue
Closes: #7155

## Description of Changes
- Changes to the community members page
  - ~~Added throttling to limit to 1 page fetch per second~~
  - Disabled the `CWTable` internal sorting– the table's internal sorting didn't match the API call's sorting, which resulted in infinite loop and unpredictable behavior
  - Fixed the search bar behavior– it was not passing the query to the API call previously

## "How We Fixed It"
- The `CWTable` has in-memory sorting, which conflicted with the `orderBy` param set for the `useSearchProfilesQuery` API call.
  - When new items were added to the list, the items were rendered in an unpredictable way, with items always rendered "above the fold", resulting infinite loop of fetching pages.
  - Solution was to disable sorting for the `CWTable` and rely on the order provided by the API response. Use a single source of truth for order.

## Test Plan
- CA test
  - Go to http://localhost:8080/ethereum/members?tab=all-members
  - Continuously scroll down for ~10 seconds– confirm that pages are fetched without rapid fire API calls
  - Use the search bar to search for profile. Confirm it works properly.

## Deployment Plan
N/A

## Other Considerations
- This is a workout that comes at the cost of disabled sorting on the members page. It's set to always sort by last activity.
- A proper fix would require making the `CWTable` a controlled component– because currently it contains sorting state internally. That means we cannot have a single source of truth for order column without refactoring the `CWTable` component, which is used in 5 different places.
  - Will create FF ticket for this.
